File: ruby/test/steam/communtiy/steam_id_tests.rb
# This code is free software; you can redistribute it and/or modify it under the
# terms of the new BSD License.
#
# Copyright (c) 2009, Sebastian Staudt

$:.push File.join(File.dirname(__FILE__), '..', '..', '..', 'lib')

require "steam/community/steam_id"
require "test/unit"

class SteamIdTests < Test::Unit::TestCase

  def test_bypass_cache
    SteamId.clear_cache
    steam_id = SteamId.new('koraktor')
    fetch_time = steam_id.fetch_time
    sleep 1
    steam_id = SteamId.new('koraktor', true, true)
    assert_operator(fetch_time, :<, steam_id.fetch_time)
  end

  def test_cache
    SteamId.clear_cache
    steam_id = SteamId.new('koraktor')
    fetch_time = steam_id.fetch_time
    assert_equal(true, SteamId.cached?(76561197961384956))
    assert_equal(true, SteamId.cached?('koraktor'))
    sleep 1
    steam_id = SteamId.new('koraktor')
    assert_equal(fetch_time, steam_id.fetch_time)
  end

  def test_case_insensitivity
    SteamId.clear_cache
    steam_id = SteamId.new('koraktor', false)
    steam_id2 = SteamId.new('Koraktor', false)
    steam_id3 = SteamId.new('KORAKTOR', false)
    assert_equal(true, SteamId.cached?('koraktor'))
    assert_equal(steam_id, steam_id2)
    assert_equal(steam_id, steam_id3)
  end

  def test_convert_steam_id_to_community_id
    steam_id64 = SteamId.convert_steam_id_to_community_id('STEAM_0:0:12345')
    assert_equal(76561197960290418, steam_id64)
  end

end

File: php/test/steam/community/SteamIdTests.php
<?php
/**
 * This code is free software; you can redistribute it and/or modify it under
 * the terms of the new BSD License.
 *
 * Copyright (c) 2009-2011, Sebastian Staudt
 *
 * @license http://www.opensource.org/licenses/bsd-license.php New BSD License
 */

error_reporting(E_ALL & ~E_USER_NOTICE);

require_once dirname(__FILE__) . '/../../../lib/steam-condenser.php';

require_once 'PHPUnit/Framework.php';

/**
 * @author     Sebastian Staudt
 * @package    steam-condenser
 * @subpackage tests
 */
class SteamIdTests extends PHPUnit_Framework_TestCase {

    public function testBypassCache() {
        SteamId::clearCache();
        $steamId = SteamId::create('koraktor');
        $fetchTime = $steamId->getFetchTime();
        sleep(1);
        $steamId = SteamId::create('koraktor', true, true);
        $this->assertGreaterThan($fetchTime, $steamId->getFetchTime());
    }

    public function testCache() {
        SteamId::clearCache();
        $steamId = SteamId::create('koraktor');
        $fetchTime = $steamId->getFetchTime();
        $this->assertTrue(SteamId::isCached('76561197961384956'));
        $this->assertTrue(SteamId::isCached('koraktor'));
        sleep(1);
        $steamId = SteamId::create('koraktor');
        $this->assertEquals($fetchTime, $steamId->getFetchTime());
    }

    public function testCaseInsensitivity() {
        SteamId::clearCache();
        $steamId = SteamId::create('koraktor', false);
        $steamId2 = SteamId::create('Koraktor', false);
        $steamId3 = SteamId::create('KORAKTOR', false, true);
        $this->assertTrue(SteamId::isCached('koraktor'));
        $this->assertEquals($steamId, $steamId2);
        $this->assertEquals($steamId, $steamId3);
    }

    public function testConvertSteamIdToCommunityId() {
        $steamId64 = SteamId::convertSteamIdToCommunityId('STEAM_0:0:12345');
        $this->assertType(PHPUnit_Framework_Constraint_IsType::TYPE_STRING,
            $steamId64);
        $this->assertEquals('76561197960290418', $steamId64);
    }
}
